src/connect.ts

```diff
@@ -330,18 +328,22 @@ export async function connectMongoClient(
   return { client, state };
 }
 
-export function oidcHasProviderFlow(uri: string, clientOptions: MongoClientOptions): boolean {
-  if (clientOptions.authMechanismProperties?.PROVIDER_NAME) {
-    return true;
+export function isHumanOidcFlow(uri: string, clientOptions: MongoClientOptions): boolean {
+  if (
+    clientOptions.authMechanism !== 'MONGODB-OIDC' ||
+    clientOptions.authMechanismProperties?.PROVIDER_NAME
+  ) {
+    return false;
   }
   let cs: ConnectionString;
   try {
     cs = new ConnectionString(uri, { looseValidation: true });
   } catch {
-    return false;
+    return true;
   }
 
-  return !!new CommaAndColonSeparatedRecord(
-    cs.typedSearchParams<MongoClientOptions>().get('authMechanismProperties')
+  const sp = cs.typedSearchParams<MongoClientOptions>();
+  return sp.get('authMechanism') === 'MONGODB-OIDC' && !new CommaAndColonSeparatedRecord(
+    sp.get('authMechanismProperties')
```

Comment:
@Anemy Looks like your suggestion from https://github.com/mongodb-js/devtools-connect/pull/23#discussion_r1188265704 is needed here because the driver seems to fail when `authMechanismProperties` are set but no auth mechanism